docs/source/common/weights_loading.rst

```diff
@@ -216,3 +179,180 @@ do the following:
+            monitor="global_step",
+            mode="max",
+            dirpath="my/path/",
+            filename="sample-mnist-{epoch:02d}-{global_step}",
+        )
+
+- When using iterative training which doesn't have an epoch, you can checkpoint at every ``N`` training steps by specifying ``every_n_training_steps=N``.
+- In case your training is interrupted for some reason, you can specify ``save_last=True`` to ensure that a checkpoint is being created at the end so that you can
+  resume it or use it for :doc:`fault-tolerant training <../advanced/fault_tolerant_training>`.
+- You can also control the interval of epochs between checkpoints using ``every_n_epochs`` between checkpoints, to avoid slowdowns.
+- By default ``ModelCheckpoint`` callback saves model weights, optimizer states, etc., but in case you have limited disk space or just need the model weights to be saved you can specify ``save_weights_only=True``.
+- Not just the training epoch/steps, you can checkpoint at regular time interval using ``train_time_interval`` argument.
+
+|
+
+:class:`~pytorch_lightning.callbacks.ModelCheckpoint` callback is preety robust and should cover 99% of the use-cases. In case you have a use-case that is not configured yet, feel free to open an issue with a feature-request on GitHub
+and Lightning Team would be happy to integrate/help integrate it.
+
+
+-----------
+
+
+***********************
+Customize Checkpointing
+***********************
+
+.. warning::
+
+    The Checkpoint IO API is experimental and subject to change.
+
+
+Lightning supports modifying the checkpointing save/load functionality through the ``CheckpointIO``. This encapsulates the save/load logic
+that is managed by the ``TrainingTypePlugin``. ``CheckpointIO`` is different from :meth:`~pytorch_lightning.core.hooks.CheckpointHooks.on_save_checkpoint`
+and :meth:`~pytorch_lightning.core.hooks.CheckpointHooks.on_load_checkpoint` methods as it determines how the checkpoint is saved/loaded rather than
+what's saved in the checkpoint.
+
+
+Built-in Checkpoint IO Plugins
+==============================
+
+.. list-table:: Built-in Checkpoint IO Plugins
+   :widths: 25 75
+   :header-rows: 1
+
+   * - Plugin
+     - Description
+   * - :class:`~pytorch_lightning.plugins.io.TorchCheckpointIO`
+     - CheckpointIO that utilizes :func:`torch.save` and :func:`torch.load` to save and load checkpoints
+       respectively, common for most use cases.
+   * - :class:`~pytorch_lightning.plugins.io.XLACheckpointIO`
+     - CheckpointIO that utilizes :func:`xm.save` to save checkpoints for TPU training strategies.
+
+
+Custom Checkpoint IO Plugin
+===========================
+``CheckpointIO`` can be extended to include your custom save/load functionality to and from a path. The ``CheckpointIO`` object can be passed to either a ``Trainer`` directly or a ``TrainingTypePlugin`` as shown below:
+
+.. code-block:: python
+
+    from pathlib import Path
+    from typing import Any, Dict, Optional, Union
```

Comment:
Can you remove the typing? It's noise for the reader and if they need that information, they would look it up in their IDE / API reference.